libcodechecker/store.py

```diff
@@ -308,6 +314,25 @@ def __get_run_name(input_list):
         return False
 
 
+def process_compile_db(compile_db):
+    """
+    Create a dict where the source file name (absolute path) is the key and
+    the correspoding compilation command is the value.
+    """
+    compile_db = os.path.abspath(compile_db)
+
+    comp_cmds = {}
+    with open(compile_db, 'r') as ccdb:
+        comp_cmds = json.load(ccdb)
+
+    processed = {}
+    for cc in comp_cmds:
+        processed[str(os.path.join(cc['directory'], cc['file']))] = \
+                str(' '.join(cc['arguments']))
```

Comment:
Newer `scan-build-py` creates such compilation command databases. The build log parser was patched to support both file formats, see the last commit 4137434e96c36557fb4612bd32ca175fe59d3267 in #610.